File: Remove Linked List Elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param head, a ListNode
    # @param val, an integer
    # @return a ListNode
    def removeElements(self, head, val):
        # Write your code here
        result = self
        result.next = head
        p = result
        q = head
        while q != None:
            if q.val == val:
                p.next = q.next
            else:
                p = p.next
        
            q = q.next
        
        return result.next

File: Remove Linked List Elements.java
Example
Given 1->2->3->3->4->5->3, val = 3, you should return the list as 1->2->4->5


/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode p = dummy;
        ListNode q = head;
        while(q!=null) {
            if(q.val == val) {
                p.next = q.next;
            } else {
                p = p.next;
            }
            q = q.next;
        }
        
        return dummy.next;
    }
}
